### ISSUE TITLE
ChoicePrompt handles irrelevant utterance as a choice because of ordinals/numbers

### ISSUE THREAD

**Issue by EdwinOtten** [microsoft/botbuilder-dotnet#2956]
**Is your feature request related to a problem? Please describe.**
If the user ignores the buttons of a ChoicePrompt and asks another (totally different) question, the bot may recognizes ordinals/numbers in the utterance and handles that as if the user made a choice. This causes problems for me in multiple cases:
1. When the utterance contains a number (which is irrelevant if the user changes subject)
2. When the utterance contains the Dutch word `een`, which translates to `one` or `a` depending on the context.
Imagine a user saying `i want to talk to a human`, and the bot handles `a` as option 1.

**Describe the solution you'd like**
I'd like to disable part of the [logic in ChoiceRecognizers](https://github.com/microsoft/botbuilder-dotnet/blob/d6f7f5aad6e99c0c407093f1c5b0d3ecb823fada/libraries/Microsoft.Bot.Builder.Dialogs/Choices/ChoiceRecognizers.cs#L53) so it won't recognize choices based on ordinals/numbers. This option could be added to [FindChoicesOptions](https://github.com/microsoft/botbuilder-dotnet/blob/040cd5499fdadcc8a3a7088a7ee106bfbdde4339/libraries/Microsoft.Bot.Builder.Dialogs/Choices/FindChoicesOptions.cs).

The V3 .NET SDK offered ChoiceOptions, which allowed us to disable these features ([source](https://stackoverflow.com/a/45028801/1644624)).
```
/// <param name="recognizeChoices">(Optional) if true, the prompt will attempt to recognize numbers in the users utterance as the index of the choice to return. The default value is "true".</param>
/// <param name="recognizeNumbers">(Optional) if true, the prompt will attempt to recognize ordinals like "the first one" or "the second one" as the index of the choice to return. The default value is "true".</param>
/// <param name="recognizeOrdinals">(Optional) if true, the prompt will attempt to recognize the selected value using the choices themselves. The default value is "true".</param>
```

Otherwise, I'd like to supply my own implementation of ChoiceRecognizers to the ChoicePrompt.

**Describe alternatives you've considered**
I have considered implementing my own ChoicePrompt and use some custom logic instead of ChoiceRecognizers to recognize choices.

**Additional context**
In V3 someone had a [similar issue](https://github.com/microsoft/botframework-sdk/issues/3115) because the titles of his Choice buttons contained numbers. Back then, @EricDahlvang [suggested](https://github.com/microsoft/botframework-sdk/issues/3115#issuecomment-437474982) to set recognizeOrdinals to false.

[enhancement]

**Comment by v-kydela** [comment]
I just want to mention that another customer recently had this issue in v4: https://stackoverflow.com/questions/58743910/how-to-have-cancel-button-in-adaptive-cards-to-be-displayed-in-web-channel-chat

A long JSON string was being sent to the dialog and being recognized as choice 1 just because the JSON had a 1 as the value of one of its properties, even though it obviously was not meant as a choice.

I'd like to suggest that in addition to disabling ordinals/numbers, perhaps some NLU could be applied to the recognition process. Right now it's very naive as you've seen, to the point where it will recognize a number anywhere in an utterance no matter what the context is.

However, I think the intention behind prompts is that if you want to allow for interruptions like "I want to talk to a human," those interruptions should be applied before the utterance ever makes it to the prompt. That would mean it's not that important for a choice prompt to be intelligent enough to know when an utterance isn't meant as a choice.

Since you seem to already have the idea of inheriting from `ChoicePrompt` in your own class, I'll assume you're not satisfied with any existing workarounds. I'm going to unassign myself so that the rest of the team can decide if they want to make any design changes here.

**Comment by benbrown** [comment]
@EricDahlvang can you take a peek at this? Is there anything we can do?

**Comment by EricDahlvang** [comment]
Good suggestion.  It seems reasonable to provide options to disable ordinal recognition in https://github.com/microsoft/botbuilder-dotnet/blob/040cd5499fdadcc8a3a7088a7ee106bfbdde4339/libraries/Microsoft.Bot.Builder.Dialogs/Choices/FindChoicesOptions.cs

**Comment by gabog** [comment]
Thanks for the suggestion, I am tagging this issue so we can reconsider it in our next iteration.

**Comment by EricDahlvang** [comment]
FindChoicesOptions now has RecognizeOrdinals and RecognizeNumbers options. This will be released with 4.8

**Comment by EdwinOtten** [comment]
@EricDahlvang that's great news! Thanks for making this happen 😃
